examples/app-dir-experiments/app/ssr/ApolloWrapper.tsx

```diff
@@ -6,21 +6,18 @@ import {
   HttpLink,
   SuspenseCache,
 } from "@apollo/client";
+import { SchemaLink } from "@apollo/client/link/schema";
 import {
   ApolloNextAppProvider,
   NextSSRInMemoryCache,
   SSRMultipartLink,
 } from "@apollo/experimental-nextjs-app-support/ssr";
 import { setVerbosity } from "ts-invariant";
+import { schema } from "../api/graphql/route";
```

Comment:
I'm wondering about how importing `schema` here could expose credentials, `schema` with resolvers would need access to the DB driver, which in turn would need some credentials, we have a check for `isServer` so `SchemaLink` is only used in the server side, but the `schema` is imported for both, meaning the bundle might end up with the driver setup in the client, and depending on how it's bundled it would have the credentials embedded somewhere, do you think I'm making a correct assumption here?

Reply:
From my experiences with the Nextjs bundler, I would assume that the `typeof window !== "undefined"` check would remove `SchemaLink` as well as `schema` from the client bundle shipped to the browser, but I'd be grateful if you could validate that further :)

Reply:
Can't validate, my setup uses Neo4j GraphQL schema, which is generated async, creating the wrapper with the schema makes it harder since the `makeClient` function would need to be async, and Next.js complains about functional components that can't be async, sorry.

Reply:
I am dumb, dunno why I didn't think about this before but to solve my issue of schema generator being async, I could just wrap `makeClient` into an async factory, declare `<ApolloWrapper>` component as async and inside the component, await the factory to get the `makeClient`, can't believe it took me this long to think about this solution...